lib/tbd/psi.rb

```diff
@@ -268,110 +419,141 @@ def gen(id = "")
-      v[:cornerconcave  ] = @set[id][:corner         ] if h[:corner         ]
-      v[:cornerconvex   ] = @set[id][:corner         ] if h[:corner         ]
-      v[:cornerconcave  ] = @set[id][:cornerconcave  ] if h[:cornerconcave  ]
-      v[:cornerconvex   ] = @set[id][:cornerconvex   ] if h[:cornerconvex   ]
-      v[:parapet        ] = @set[id][:parapet        ] if h[:parapet        ]
-      v[:parapetconcave ] = @set[id][:parapet        ] if h[:parapet        ]
-      v[:parapetconvex  ] = @set[id][:parapet        ] if h[:parapet        ]
-      v[:parapetconcave ] = @set[id][:parapetconcave ] if h[:parapetconcave ]
-      v[:parapetconvex  ] = @set[id][:parapetconvex  ] if h[:parapetconvex  ]
-      v[:party          ] = @set[id][:party          ] if h[:party          ]
-      v[:partyconcave   ] = @set[id][:party          ] if h[:party          ]
-      v[:partyconvex    ] = @set[id][:party          ] if h[:party          ]
-      v[:partyconcave   ] = @set[id][:partyconcave   ] if h[:partyconcave   ]
-      v[:partyconvex    ] = @set[id][:partyconvex    ] if h[:partyconvex    ]
-      v[:grade          ] = @set[id][:grade          ] if h[:grade          ]
-      v[:gradeconcave   ] = @set[id][:grade          ] if h[:grade          ]
-      v[:gradeconvex    ] = @set[id][:grade          ] if h[:grade          ]
-      v[:gradeconcave   ] = @set[id][:gradeconcave   ] if h[:gradeconcave   ]
-      v[:gradeconvex    ] = @set[id][:gradeconvex    ] if h[:gradeconvex    ]
-      v[:balcony        ] = @set[id][:balcony        ] if h[:balcony        ]
-      v[:balconyconcave ] = @set[id][:balcony        ] if h[:balcony        ]
-      v[:balconyconvex  ] = @set[id][:balcony        ] if h[:balcony        ]
-      v[:balconyconcave ] = @set[id][:balconyconcave ] if h[:balconyconcave ]
-      v[:balconyconvex  ] = @set[id][:balconyconvex  ] if h[:balconyconvex  ]
-      v[:rimjoist       ] = @set[id][:rimjoist       ] if h[:rimjoist       ]
-      v[:rimjoistconcave] = @set[id][:rimjoist       ] if h[:rimjoist       ]
-      v[:rimjoistconvex ] = @set[id][:rimjoist       ] if h[:rimjoist       ]
-      v[:rimjoistconcave] = @set[id][:rimjoistconcave] if h[:rimjoistconcave]
-      v[:rimjoistconvex ] = @set[id][:rimjoistconvex ] if h[:rimjoistconvex ]
+      v               = {} # PSI-value (W/K per linear meter)
+      v[:joint      ] = 0; v[:transition        ] = 0; v[:fenestration     ] = 0
+      v[:head       ] = 0; v[:headconcave       ] = 0; v[:headconvex       ] = 0
+      v[:sill       ] = 0; v[:sillconcave       ] = 0; v[:sillconvex       ] = 0
+      v[:jamb       ] = 0; v[:jambconcave       ] = 0; v[:jambconvex       ] = 0
+      v[:corner     ] = 0; v[:cornerconcave     ] = 0; v[:cornerconvex     ] = 0
+      v[:parapet    ] = 0; v[:parapetconcave    ] = 0; v[:parapetconvex    ] = 0
+      v[:roof       ] = 0; v[:roofconcave       ] = 0; v[:roofconvex       ] = 0
```

Comment:
Introduces an ASHRAE-based distinction between wall-to-roof intersections (to finalize later):

- as parapets or overhangs
- all other cases